### ISSUE TITLE
Can we show or hide specific time slots in week view

### ISSUE THREAD

**Issue by Rishabh-Jain-Rj** [jquense/react-big-calendar#2415]
### Clear and concise description of the problem

hello! This is Rishabh !
i am trying to show some specific time slots in a week view but i am not able to find the correct solution. I have tried the min and max props and these are helpful , if we wanted to show a range of timeslots but my requirement is to show specific timeslots ( in my case i wanted to show only 2-3 timeslots but these are not in consecutive order) so in that case min and max props approach failed.

To achieve this thing i also tried to play with html i made all timeslots display : none except the timeslots in which event happend

    let timeslotColumns = document.querySelectorAll(".rbc-time-column");
    timeslotColumns.forEach((e) => {
      let childrens = e.children;
      Array.from(childrens).forEach((e, i) => {
        if (e.classList.contains("rbc-timeslot-group"))
          e.style.display = "none";
        e.classList.add(`timeslot-${i}`);
      });
    });
    const a = document.querySelectorAll(".timeslot-17");
    console.log(a);
    Array.from(a).forEach((e) => {
      e.style.display = "block";
    });

but this logic failed because after doing this the event is not showing properly 

![image](https://github.com/jquense/react-big-calendar/assets/102375298/3cfcb226-6ed1-446d-8a20-9a491a2a8f13)

can anyone help me to solve this problem

### Suggested solution

              min={minTime}
              max={maxTime}

### Validations

- [X] Read the [docs](http://jquense.github.io/react-big-calendar/examples/index.html).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Would you like to open a PR for this feature?

- [ ] I'm willing to open a PR

**Comment by cutterbl** [comment]
@Rishabh-Jain-Rj What, exactly, is it you are trying to accomplish with these 'slots' you are trying to display? Give us the use case of what it is you are trying to achieve.

**Comment by utkarshsystem** [comment]
We have huge data of students which we show in weekly view. We want to merge rows and hide rows to get more spacing. We are looking for a feature where we can provide custom time slots only to display weekly views. 
For example, 
Use Case 1: by default, all time slots are hidden, and if we add Mon 12-1 PM, it will show only 1 row of time slots which we can achieve using min and max. 
Use Case 2: We added Tues 2-3 PM now we want to show only 2-time slots from the left panel 12-1, and 2-3 PM. The benefit of doing this we have a custom interface in the cell which we really need a big space.

I hope I have clarified the use cases.

**Comment by cutterbl** [comment]
@utkarshsystem No, I'm afraid I still do not understand what it is you are ultimately trying to accomplish.

If you set `timeslot: 1` and `step: 60`, then there are no grid lines in an hour range. That said, the event you place in that area would only take up all of that height if the start and end also covered that entire hour long block.

In my own application, `events` can be as short as 15 minute, or last multiple days. I only show abbreviated info related to my `event` (an appointment) (like the person it is for and the start and end times) in the calendar/TimeGrid view. For my users to get an expanded view of all of the `event`'s (appointment's) details, the user selects the `event`, which shows that expanded detail in a side panel. That detail could be as long as I want, and I have very detailed display logic around those details, completely separate from all the Big Calendar logic. A Master/Detail view, if you will.

**Comment by cutterbl** [comment]
@Rishabh-Jain-Rj Currently there is no native support for such a function. You may have to write a custom view to accomplish what you're asking.

**Comment by usmanabdurrehman** [comment]
@Rishabh-Jain-Rj I get what you are saying. He basically wants to show some specific timeslots based on if they have events or not. Let's say the usual timeslot range is 12:00 PM to 12:00 AM but events are present only for 02:00 PM to 03:00 PM and 08:00 PM to 09:00 PM so he would want to show only those slots with maybe interpolation dots going on showing that there are other slots between them but they are not visible or something. @cutterbl Can I try to implement it?

**Comment by cutterbl** [comment]
@usmanabdurrehman You can try, but you would definitely need a custom view, and probably need some custom layout algorithm or something...

**Comment by usmanabdurrehman** [comment]
@cutterbl Cool. Let me see.

**Comment by KreisonReis** [comment]
I have the same question
